doc/devel/development.md

```diff
@@ -0,0 +1,1932 @@
+| `attributes`          | `active`             | List of active attributes of the device                                                                                                                                                                                                                                   | `[ { "name": "attr_name", "type": "Text" } ]` |
+| `lazy`                | `lazy`               | List of lazy attributes of the device                                                                                                                                                                                                                                     | `[ { "name": "attr_name", "type": "Text" } ]` |
+| `commands`            | `commands`           | List of commands of the device                                                                                                                                                                                                                                            | `[ { "name": "attr_name", "type": "Text" } ]` |
+| `internal_attributes` | `internalAttributes` | List of internal attributes with free format for specific IoT Agent configuration                                                                                                                                                                                         | LWM2M mappings from object URIs to attributes |
+| `static_attributes`   | `staticAttributes`   | List of static attributes to append to the entity. All the updateContext requests to the CB will have this set of attributes appended.                                                                                                                                    | `[ { "name": "attr_name", "type": "Text" } ]` |
+| `expressionLanguage`  | `expresionLanguage`  | optional boolean value, to set expression language used to compute expressions, possible values are: legacy or jexl. When not set or wrongly set, legacy is used as default value.                                                                                        |
+| `explicitAttrs`       | `explicitAttrs`      | optional field to support selective ignore of measures so that IOTA doesn’t progress. See details in [specific section](../api.md#explicitly-defined-attributes-explicitattrs)                                                                                            | (see details in specific section)             |
+| `ngsiVersion`         | `ngsiVersion`        | optional string value used in mixed mode to switch between **NGSI-v2** and **NGSI-LD** payloads. The default is `v2`. When not running in mixed mode, this field is ignored.                                                                                              | `v2/ld`                                       |
+
+## Data mapping plugins
```

Comment:
"plugins" is mentioned several times in this documentation...

However plugins has been removed in 3.3.0 (or are in the process to be removed in 3.4.0). Is still correct using "plugin" in documentation?

Reply:
I think this should be done later, a specific task looking for all outdated functionalities removed in latest versions (like append mode, plugins etc). I would also affect to other documents.

Reply:
In my opinion, a general review of the documentation to remove obsolete pieces is a good idea. However, that doesn't preclude that if we spot something obsolete in a review, we take the opportunity to fix that *assuming that the fix is easy*, due to:

* That will simplify the final review task
* Documentation is large and something could be missed in the final review task (we are humans :). If we fix things as we found it, we minimize the probability of leaving obsolete pieces after the final review task

Reply:
After discussion, this is going to be done in a subsequent PR after modifying the doc structure over all documents. This would not increase the file scope of this PR. Noted here: https://github.com/telefonicaid/iotagent-node-lib/issues/1329#issue-1643605738